backend/src/schema/resolvers/posts.js

```diff
@@ -181,4 +182,46 @@ export default {
       return emoted
     },
   },
+  Post: {
+    ...Resolver('Post', {
+      hasMany: {
+        tags: '-[:TAGGED]->(related:Tag)',
+        categories: '-[:CATEGORIZED]->(related:Category)',
+        comments: '<-[:COMMENTS]-(related:Comment)',
+        shoutedBy: '<-[:SHOUTED]-(related:User)',
+        emotions: '<-[related:EMOTED]',
+      },
+      hasOne: {
+        author: '<-[:WROTE]-(related:User)',
+        disabledBy: '<-[:DISABLED]-(related:User)',
+      },
+      count: {
+        shoutedCount:
+          '<-[:SHOUTED]-(related:User) WHERE NOT related.deleted = true AND NOT related.disabled = true',
+        emotionsCount: '<-[related:EMOTED]-(:User)',
+      },
+      boolean: {
+        shoutedByCurrentUser:
+          '<-[:SHOUTED]-(u:User {id: $cypherParams.currentUserId}) RETURN COUNT(u) >= 1',
+      },
+    }),
+    relatedContributions: async (parent, params, context, resolveInfo) => {
+      if (typeof parent.relatedContributions !== 'undefined') return parent.relatedContributions
+      const { id } = parent
+      const statement = `
+      MATCH (p:Post {id: $id})-[:TAGGED|CATEGORIZED]->(categoryOrTag)<-[:TAGGED|CATEGORIZED]-(post:Post)
+      RETURN DISTINCT post
+      LIMIT 10
+      `
+      let relatedContributions
+      const session = context.driver.session()
+      try {
+        const result = await session.run(statement, { id })
+        relatedContributions = result.records.map(r => r.get('post').properties)
+      } finally {
+        session.close()
+      }
+      return relatedContributions
+    },
+  },
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-08-30T12:29:45Z" title="Friday, August 30th 2019, 2:29:45 pm +02:00">Aug 30, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -179,4 +180,46 @@ export default {
       return emoted
     },
   },
+  Post: {
+    ...Resolver('Post', {
+      hasMany: {
+        tags: '-[:TAGGED]->(related:Tag)',
+        categories: '-[:CATEGORIZED]->(related:Category)',
+        comments: '<-[:COMMENTS]-(related:Comment)',
+        shoutedBy: '<-[:SHOUTED]-(related:User)',
+        emotions: '<-[related:EMOTED]',
+      },
+      hasOne: {
+        author: '<-[:WROTE]-(related:User)',
+        disabledBy: '<-[:DISABLED]-(related:User)',
+      },
+      count: {
+        shoutedCount:
+          '<-[:SHOUTED]-(related:User) WHERE NOT related.deleted = true AND NOT related.disabled = true',
+        emotionsCount: '<-[related:EMOTED]-(:User)',
+      },
+      boolean: {
+        shoutedByCurrentUser:
+          '<-[:SHOUTED]-(u:User {id: $cypherParams.currentUserId}) RETURN COUNT(u) >= 1',
+      },
+    }),
+    relatedContributions: async (parent, params, context, resolveInfo) => {
+      if (typeof parent.relatedContributions !== 'undefined') return parent.relatedContributions
+      const { id } = parent
+      const statement = `
+      MATCH (p:Post {id: $id})-[:TAGGED|CATEGORIZED]->(categoryOrTag)<-[:TAGGED|CATEGORIZED]-(post:Post)
+      RETURN DISTINCT post
+      LIMIT 10
+      `
+      let relatedContributions
+      const session = context.driver.session()
+      try {
+        const result = await session.run(statement, { id })
+        relatedContributions = result.records.map(r => r.get('post').properties)
+      } finally {
+        session.close()
+      }
+      return relatedContributions
+    },
+  },
```


so, I don't need to implement this anymore... I was quite keen on adding this code, but I was in pairing sessions with other people the last two days, and I understand it was blocking you...